File: app/src/main/java/com/cadnative/firevisioniptv/TvSetupActivity.java
package com.cadnative.firevisioniptv;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

/**
 * Setup Activity for Fire TV TIF Integration
 * This launches when users navigate to Settings > Live TV > Sync Sources
 * It can redirect to your existing PairingActivity or handle setup directly
 */
public class TvSetupActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Option 1: Redirect to existing PairingActivity
        Intent pairingIntent = new Intent(this, PairingActivity.class);
        pairingIntent.putExtra("source", "tv_setup");
        startActivityForResult(pairingIntent, 1);

        // Option 2: Show simple message and sync channels
        // Toast.makeText(this, "Syncing FireVision IPTV channels...", Toast.LENGTH_SHORT).show();
        // syncChannels();
        // setResult(RESULT_OK);
        // finish();
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (requestCode == 1) {
            if (resultCode == RESULT_OK) {
                // Pairing successful, sync channels
                Toast.makeText(this, "Device paired! Syncing channels...", Toast.LENGTH_SHORT).show();
                syncChannels();
            }
            setResult(resultCode);
            finish();
        }
    }

    private void syncChannels() {
        // Trigger channel sync via ChannelManager
        ChannelManager channelManager = new ChannelManager(this);
        channelManager.syncChannelsToTif();
    }
}

File: app/src/main/java/com/cadnative/firevisioniptv/TvSetupActivity.kt
package com.cadnative.firevisioniptv

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.activity.ComponentActivity
import androidx.activity.result.contract.ActivityResultContracts
import androidx.lifecycle.lifecycleScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class TvSetupActivity : ComponentActivity() {

    private val pairingLauncher = registerForActivityResult(
        ActivityResultContracts.StartActivityForResult()
    ) { result ->
        if (result.resultCode == RESULT_OK) {
            Toast.makeText(this, "Device paired! Syncing channels...", Toast.LENGTH_SHORT).show()
            syncChannels()
        }
        setResult(result.resultCode)
        finish()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val pairingIntent = Intent(this, PairingActivity::class.java).apply {
            putExtra("source", "tv_setup")
        }
        pairingLauncher.launch(pairingIntent)
    }

    private fun syncChannels() {
        lifecycleScope.launch(Dispatchers.IO) {
            try {
                val channelManager = ChannelManager.create(this@TvSetupActivity)
                channelManager.syncChannelsToTif()
            } catch (e: Exception) {
                android.util.Log.e("TvSetupActivity", "Error syncing channels", e)
            }
        }
    }
}
